symengine/sympy_compat.py

```
@@ -43,6 +47,35 @@ def __new__(cls, i):
         return symengine.Integer(i)
 
 
+class Float(Number):
+    _classes = (symengine.RealDouble, symengine.RealMPFR)
+
+    def __new__(cls, num, dps=None, prec=None, precision=None):
+        if dps is not None and precision is not None:
+            raise ValueError('Both decimal and binary precision supplied. '
+                             'Supply only one. ')
+        if dps is None and precision is None:
+            dps = 15
+            if isinstance(num, Float):
+                return num
+        elif precision == '' and dps is None or precision is None and dps == '':
+            if not isinstance(num, (str, unicode)):   
+                raise ValueError('The null string can only be used when '
+                'the number to Float is passed as a string or an integer.')
+        if precision is None or precision == '':
+            precision = dps_to_prec(dps)
+        if HAVE_SYMENGINE_MPFR:
+            if precision > 53:
+                return RealMPFR(str(num), precision)    
+            else:
+                return RealDouble(float(str(num)))
+        else:
+            RealDouble(float(str(num)))
```

This should throw an error if `precision > 53`